pandas/core/tools/datetimes.py

```diff
@@ -534,25 +534,19 @@ def _to_datetime_with_unit(arg, unit, name, tz, errors: Optional[str]) -> Index:
     # GH#30050 pass an ndarray to tslib.array_with_unit_to_datetime
     # because it expects an ndarray argument
     if isinstance(arg, IntegerArray):
-        result = arg.astype(f"datetime64[{unit}]")
+        res = arg.astype(f"datetime64[{unit}]")
```

Comment:
nitpick: res-> arr (I think once the type is added for `arg` in the signature, we will get an `Incompatible types in assignment` in `arg = getattr(arg, "_values", arg)` anyway.